benchmark/domain/synthetic_3.py

```diff
@@ -0,0 +1,92 @@
+"""Synthetic dataset. Custom parabolic test with irrelevant parameters."""
+
+from uuid import UUID
+
+from numpy import pi, sin, sqrt
+from pandas import DataFrame
+
+from baybe.campaign import Campaign
+from baybe.objective import SingleTargetObjective
+from baybe.parameters import NumericalContinuousParameter, NumericalDiscreteParameter
+from baybe.recommenders import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget, TargetMode
+from benchmark.definition import Benchmark
+
+
+def lookup_synthetic_3(x: float, y: float, z: int) -> float:
+    """Synthetic dataset.
+
+        Number of Samples            inf
+        Dimensionality                 3
+        Features:
+            x   continuous [-2*pi, 2*pi]
+            y   continuous [-2*pi, 2*pi]
+            z   discrete {1,2,3,4}
+        Targets:
+            output   continuous
+    Best Value 4.09685
+    """
+    if z == 1:
+        return sin(x) * (1 + sin(y))
+    if z == 2:
+        return x * sin(0.9 * x) + sin(x) * sin(y)
+    if z == 3:
+        return sqrt(x + 8) * sin(x) + sin(x) * sin(y)
+    if z == 4:
+        return x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y)
+
+    return 0.0
+
+
+def synthetic_3() -> tuple[DataFrame, dict[str, str]]:
+    """Synthetic dataset. Custom parabolic test with irrelevant parameters."""
+    synthetic_3_continues = [
+        NumericalContinuousParameter("x", (-2 * pi, 2 * pi)),
+        NumericalContinuousParameter("y", (-6.283185, 6.283185)),
+        NumericalDiscreteParameter("z", (1, 2, 3, 4)),
+    ]
+    print(NumericalContinuousParameter("x", (-2 * pi, 2 * pi)).to_json())
+    objective = SingleTargetObjective(
+        target=NumericalTarget(name="output", mode=TargetMode.MAX)
+    )
+
+    campaign = Campaign(
+        searchspace=SearchSpace.from_product(parameters=synthetic_3_continues),
+        objective=objective,
+    )
+    campaign_rand = Campaign(
+        searchspace=SearchSpace.from_product(parameters=synthetic_3_continues),
+        recommender=RandomRecommender(),
+    )
+
+    batch_size = 5
+    n_doe_iterations = 30
+    n_mc_iterations = 50
+
+    metadata = {
+        "n_doe_iterations": str(n_doe_iterations),
+        "batch_size": str(batch_size),
+        "n_mc_iterations": str(n_mc_iterations),
+    }
+
+    scenarios = {
+        "Default Recommender": campaign,
+        "Random Baseline": campaign_rand,
+    }
+    return simulate_scenarios(
+        scenarios,
+        lookup_synthetic_3,
+        batch_size=batch_size,
+        n_doe_iterations=n_doe_iterations,
+        n_mc_iterations=n_mc_iterations,
+        impute_mode="error",
+    ), metadata
+
+
+benchmark_synthetic_3 = Benchmark(
+    name="Synthetic dataset 3. Three dimensional.",
+    identifier=UUID("4e131cb7-4de0-4900-b993-1d7d4a194532"),
```

Comment:
I guess this is just some random `str` here? Couldn't unique identifiers be set automatically?

Reply:
Yes, if it is not set, a unique ID is generated automatically. The intention was to have an identifier for the test that can be used in the CLI for backtracking purposes, since other identifiers may change (like the title) or are too large to print (like the result).

Reply:
Discussed the necessity of setting those manually, hence resolving.

Reply:
I was also wondering this, so im registering interest here to be enlightened in the dev meeting :)

if this is something temporary for testing also other parts of the entire project, please indicate that via comment